Set12Simulator/ability.py

```diff
@@ -57,2838 +57,184 @@ def apply_attack_cooldown(champion, halved=True):
     champion.add_que('clear_idle', (1 / champion.AS * 1000) / halver)
 
 
-# Aatrox pulls some of the farthest enemies toward himself, then slams the ground in front of himself,
-# dealing magic damage to all enemies hit.
-def aatrox(champion):
-    champion.add_que('change_stat', -1, None, 'ability_active', True)
-    default_ability_calls(champion)
-
-    champion.add_que('change_stat', stats.ABILITY_LENGTH[champion.name], None, 'ability_active', False)
-
-    enemies = field.find_enemies(champion)
-
-    enemy_amount = len(enemies)
-    enemies_pulled = stats.ABILITY_TARGETS[champion.name][champion.stars]
-
-    # pull max amount of targets. for example if 3 star aatrox wants to pull 5 but there's 3 alive, pull 2 instead
-    if enemy_amount < enemies_pulled + 1:
-        enemies_pulled = enemy_amount - 1
-    for i in range(1, enemies_pulled + 1):
-
-        free_hexes = []
-        target_neighbors = field.find_neighbors(champion.target.y, champion.target.x)
-
-        for n in target_neighbors:
-            c = field.coordinates[n[0]][n[1]]
-            if c is None:
-                free_hexes.append(n)
-
-        # break if there's no more space around the target unit. this should be implemented better
-        if len(free_hexes) == 0:
-            break
-
-        for h in free_hexes:
-            d = field.distance({'y': enemies[-i][0].y, 'x': enemies[-i][0].x}, {'y': h[0], 'x': h[1]}, False)
-            h.append(d)
-        free_hexes.sort(key=lambda x: x[2])
-
-        enemies[-i][0].move(free_hexes[0][0], free_hexes[0][1], True)
-
-    champion.add_que('execute_function', stats.ABILITY_LENGTH[champion.name],
-                     [aatrox_ability, {'y': champion.target.y, 'x': champion.target.x}])
-
-
-def aatrox_ability(champion, data):
-    neighbors = field.find_neighbors(data['y'], data['x'])
-    neighbors.append([data['y'], data['x']])
-
-    c = field.coordinates
-    for n in neighbors:
-        if c[n[0]][n[1]] and c[n[0]][n[1]].team != champion.team and c[n[0]][n[1]].champion:
-            champion.spell(c[n[0]][n[1]], stats.ABILITY_DMG[champion.name][champion.stars])
-
-    apply_attack_cooldown(champion)
-
 
 def ahri(champion):

```

Comment:
This is good but it's only half of the ticket. Repeat the process for the stats and item stats. Anywhere where the names of the previous set are used.